src/m365/spo/commands/roledefinition/roledefinition-add.ts

```diff
@@ -0,0 +1,134 @@
+import { Logger } from '../../../../cli';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { validation } from '../../../../utils';
+import SpoCommand from '../../../base/SpoCommand';
+import { BasePermissions, PermissionKind } from '../../base-permissions';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  name: string;
+  description?: string;
+}
+
+class SpoRoleDefinitionAddCommand extends SpoCommand {
+  public get name(): string {
+    return commands.ROLEDEFINITION_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds a new roledefinition to web';
+  }
+
+  protected get permissionsKindMap(): string[] {
+    const result: string[] = [];
+
+    for (const kind in PermissionKind) {
+      if (typeof PermissionKind[kind] === 'number') {
+        result.push(kind);
+      }
+    }
+    return result;
+  }
+  
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        rights: args.options.rights,
+        description: (!(!args.options.description)).toString() 
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '-n, --name <name>'
+      },
+      {
+        option: '-d, --description [description]'
+      },
+      {
+        option: '--rights [rights]',
+        autocomplete: this.permissionsKindMap
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.rights) {
+          const rights = args.options.rights.split(',');
+    
+          for (const item of rights) {
+            const kind: PermissionKind = PermissionKind[(item.trim() as keyof typeof PermissionKind)];
+    
+            if (!kind) {
+              return `Rights option '${item}' is not recognized as valid PermissionKind choice. Please note it is case sensitive`;
```

Comment:
Could we look into the possibilty to make this case-insensitive? I think it would the be easier to use if it's case-insensitive.

Reply:
this might be a bit tricky as the PermissionKind enum and this method is not only used here but also in `customeaction add/set` and maybe we should align this approach in all places at the same time 🤔 What would you suggest?

Reply:
Good point, this is something we beter expand towards a new issue. For now, let's leave it as is. Great job @Adam-it! 🚀

Reply:
Ok so all other comments were done. I will mark this as resolved and mark this PR as ready for review